src/Libs/GLib-2.0/Internal/SynchronizationContext.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace GLib.Internal;
+
+public sealed class GLibSynchronizationContext : SynchronizationContext
+{
+    #region Fields
+
+    // We need a static variable here instead of a static method because we need
+    // to keep the delegate object to stay alive for the application lifetime. We
+    // don't need a GCHandle for the delegate here because a static is enough to
+    // keep the delegate alive. See https://stackoverflow.com/q/75223488/1829232
+    // for a similar case.
+    private static readonly SourceFunc IdleCallback = data =>
+    {
+        var gch = GCHandle.FromIntPtr(data);
+        var action = (Action)gch.Target!;
+
+        action();
+
+        return false;
+    };
+
+    // According to https://docs.gtk.org/glib/main-loop.html#memory-management-of-sources
+    // we should not free the data in the idle callback because it is stored in a GSource
+    // object, which is still alive when the idle callback is returned. Instead, use
+    // either g_source_remove() or GDestroyNotify.
+    private static readonly DestroyNotify DestroyCallback = data =>
+    {
+        GCHandle.FromIntPtr(data).Free();
+    };
+
+    #endregion
+
+    #region Methods
+
+    public override SynchronizationContext CreateCopy()
+        => new GLibSynchronizationContext();
+
+    public override void Post(SendOrPostCallback d, object? state)
+        => ScheduleAction(() => d(state));
+
+    public override void Send(SendOrPostCallback d, object? state)
+    {
+        using var executed = new ManualResetEvent(false);
```

Comment:
Does it make sense to use [ManualResetEventSlim](https://learn.microsoft.com/de-de/dotnet/api/system.threading.manualreseteventslim?view=net-6.0)?

Reply:
I was thinking to use `ManualResetEventSlim` in the first place but the document said:

> In .NET Core and .NET 5+, the default spin-waiting duration is short: on the order of 10s of microseconds, depending on platform and processor. If you expect wait times to be much longer than that, you can still use this class instead of ManualResetEvent (perhaps configured with less or no spin-waiting). **However, the performance benefit would likely be only marginal**.

So I decided to use `ManualResetEvent` instead because the callback will get executed only when idle and I'm not sure how long it will need to wait. Let me know if you still want to use `ManualResetEventSlim` so I can update this code.

Reply:
I think we can start with `ManualResetEventSlim`. Just from a gut feeling: I don't expect "much longer wait times" as the main loop has to render the ui as fast as possible to keep the UI responsive.

To ensure the behaviour i think we should have some unit tests.